draft-ietf-rtcweb-jsep.xml

```diff
@@ -3163,6 +3164,935 @@ a=end-of-candidates
+            raddr 11.22.33.44 rport 52546
+a=end-of-candidates
+
+m=video 52546 UDP/TLS/RTP/SAVPF 100 101
+c=IN IP4 11.22.33.44
+a=rtcp:52546 IN IP4 11.22.33.44
+a=mid:v1
+a=recvonly
+a=rtpmap:100 VP8/90000
+a=rtpmap:101 rtx/90000
+a=fmtp:101 apt=100
+a=ice-ufrag:ATEn1v9DoTMB9J4r
+a=ice-pwd:AtSK0WpNtpUjkY4+86js7ZQl
+a=ice-options:trickle
+a=fingerprint:sha-256
+              19:E2:1C:3B:4B:9F:81:E6:B8:5C:F4:A5:A8:D8:73:04
+             :BB:05:2F:70:9F:04:A9:0E:05:E9:26:33:E8:70:88:A2
+a=setup:passive
+a=rtcp-mux
+a=rtcp-rsize
+a=extmap:2 urn:ietf:params:rtp-hdrext:sdes:mid
+a=rtcp-fb:100 ccm fir
+a=rtcp-fb:100 nack
+a=rtcp-fb:100 nack pli
+a=candidate:109270923 1 udp 2122194687 192.168.1.2 51556 typ host
+a=candidate:4036177503 1 udp 1685987071 11.22.33.44 52546 typ srflx
+          raddr 192.168.1.2 rport 51556
+a=candidate:3671762466 1 udp 41819903 22.33.44.55 61405 typ relay
+          raddr 11.22.33.44 rport 52546
+a=end-of-candidates
+
+m=video 52546 UDP/TLS/RTP/SAVPF 100 101
+c=IN IP4 11.22.33.44
+a=rtcp:52546 IN IP4 11.22.33.44
+a=mid:v2
+a=recvonly
+a=rtpmap:100 VP8/90000
+a=rtpmap:101 rtx/90000
+a=fmtp:101 apt=100
+a=ice-ufrag:ATEn1v9DoTMB9J4r
+a=ice-pwd:AtSK0WpNtpUjkY4+86js7ZQl
+a=ice-options:trickle
+a=fingerprint:sha-256
+              19:E2:1C:3B:4B:9F:81:E6:B8:5C:F4:A5:A8:D8:73:04
+             :BB:05:2F:70:9F:04:A9:0E:05:E9:26:33:E8:70:88:A2
+a=setup:passive
+a=rtcp-mux
+a=rtcp-rsize
+a=extmap:2 urn:ietf:params:rtp-hdrext:sdes:mid
+a=rtcp-fb:100 ccm fir
+a=rtcp-fb:100 nack
+a=rtcp-fb:100 nack pli
+a=candidate:109270923 1 udp 2122194687 192.168.1.2 51556 typ host
+a=candidate:4036177503 1 udp 1685987071 11.22.33.44 52546 typ srflx
+            raddr 192.168.1.2 rport 51556
+a=candidate:3671762466 1 udp 41819903 22.33.44.55 61405 typ relay
+            raddr 11.22.33.44 rport 52546
+a=end-of-candidates
+         ]]></artwork>
+       </figure>
+     </t>
+    </section>
+    -->
+
+    <section title="Inactive Media Answer Example" anchor="sec.inactive-examples" >
+
+      <t> In this example, browser A sends an offer to browser
+      B with an offer for data, audio, and two video tracks. Browser B immediately
+      responds with an answer that accepts the data channel but marks the rest
+      as inactive. This allows the browser to start negotiating ICE and DTLS
+      immediately. B then takes some time, perhaps prompting the user to
+      answer the call or for permission to use the microphone and/or camera.
+      At some future point in time B, sends a second offer which enables the
+      audio and video streams. The advantage of such a
+      flow is that as soon as the first answer is received, the browser can
+      proceed with ICE and DTLS negotiation and have the transport set up and
+      working. If the transport setup completes before the second offer is
+      sent -- and bundle is used -- then media can be transmitted immediately
+      upon completion of the offer/answer exchange. This technique allows
```

Comment:
I don't get that impression from that paragraph. Two arguments for my interpretation below.

Argument #1: Let's first agree that the callee could return a a=sendonly answer to the caller (indicating that the incoming stream, from the callee's perspective, is disabled). The offer was a=sendrecv, so returning a=sendonly in the answer is clearly legal. And in this state, the callee could still send media any time it wanted to, and sending a new offer with a=sendrecv would indicate that now the *caller* should send media.

This is essentially what we want, except for the fact that providing a=sendonly to setLocalDescription on the callee side will result in the callee sending media prematurely. We could try to prevent that by setting a=inactive on the callee setLocal, but sending a=sendonly to the caller in the first answer; the second offer from the callee would then change this to a=sendrecv and media would flow in both directions. 

However, it would certainly be simplest to just allow a=inactive to be used for both the callee setLocal and sent to the caller in the answer (where it is used in setRemote).

Argument #2: changing from a=inactive to a=sendrecv is essentially ending local and remote hold (as described in RFC3264, S 8.4). My reading of 8.4 is that the update of the directionality attribute is advisory; one doesn't need to wait for the answer to start/stop sending media.